docs/develop/testing/speech-service-testing.md

```diff
@@ -0,0 +1,84 @@
+# Writing and Running Tests for Speech Service
+
+This guide outlines the process for writing, running, and debugging tests for the Speech Service.
+
+## Adding new tests
+
+When developing new features for Speech Service, it's crucial to add corresponding unit tests. These tests should be placed in the following directory:
+`optional_components/speech_api/tests`.
+
+## Running unit tests
+
+**Environment Setup**
+
+Please use the `aaq-speech` environment to run speech tests. You can create this environment using the following command from the root directory:
+
+```shell
+make fresh-env-speech
+```
+
+## Executing Tests
+
+To run the tests, use the following command:
+
+```shell
+make tests
+```
```

Comment:
Let's first mention to "navigate to the speech service with `cd optional_components/speech_api`" and put the `make fresh-env-speech` inside the makefile there only?